rust/parquet/src/arrow/schema.rs

```diff
@@ -55,40 +55,75 @@ pub fn parquet_to_arrow_schema_by_columns<T>(
 where
     T: IntoIterator<Item = usize>,
 {
-    let mut base_nodes = Vec::new();
-    let mut base_nodes_set = HashSet::new();
-    let mut leaves = HashSet::new();
-
-    for c in column_indices {
-        let column = parquet_schema.column(c).self_type() as *const Type;
-        let root = parquet_schema.get_column_root(c);
-        let root_raw_ptr = root as *const Type;
-
-        leaves.insert(column);
-        if !base_nodes_set.contains(&root_raw_ptr) {
-            base_nodes.push(root);
-            base_nodes_set.insert(root_raw_ptr);
+    let mut metadata = parse_key_value_metadata(key_value_metadata).unwrap_or_default();
```

Comment:
@sunchao the main change here is that if the `ARROW:schema` exists, we try to decode it and pass it back as the schema to read.
Unlike C++, if there's a problem reading the schema, we do not return an error, but continue to reconstruct the schema with some information loss. The error is logged to console.

Reply:
👍